### PR TITLE
make animation of closing announcement smooth

### PR DESCRIPTION
**Pull request by upacyxou**
- smooth animation adde

Now transition goes on `height`, but not on `translate-y`, so the announcement is moving with DOM and it looks way smoother

https://user-images.githubusercontent.com/57546893/183931961-4fe7223e-4def-497c-b202-7e491aca5d4c.mp4

### CODE REVIEW

File: src/components/Announcement.tsx [hunk 1]
```diff
@@ -12,27 +12,36 @@ import classnames, {
...
+    padding(animate ? 'px-0' : 'px-6', animate ? 'py-0' : 'py-4'),
     display('flex'),
     alignItems('items-center'),
     justifyContent('justify-center'),
     fontWeight('font-bold'),
     fontSize('text-sm'),
     lineHeight('leading-5'),
-    transitionProperty('transition-transform'),
-    translate(animate ? '-translate-y-full' : undefined)
+    height(animate ? 'h-0' : undefined),
```
Comment (T-Damer) [inline src/components/Announcement.tsx:42]: Same here `{'h-0': animate}`

File: src/components/Announcement.tsx [hunk 2]
```diff
@@ -12,27 +12,36 @@ import classnames, {
...
+    transitionProperty('transition-height', 'transition-padding'),
```
Comment (T-Damer) [inline src/components/Announcement.tsx:43]: Better do `transition-all`, take a look here: https://developer.mozilla.org/en-US/docs/Web/CSS/transition-property

Comment (T-Damer) [reply src/components/Announcement.tsx:43]: When you specify multiple transitions from tailwind, you specify many things like timing-function, duration: https://tailwindcss.com/docs/transition-property, they will just exist as dead code in CSS

File: tailwind.config.js [hunk 3]
```diff
@@ -139,11 +139,15 @@ module.exports = {
...
         '-5.5': '-0.35rem',
         '-7': '-0.45rem',
-        '-full': '-100%',
       },
       backgroundImage: {
         noise: 'url("/img/noise50.png")',
       },
+      transitionProperty: {
+        height: 'height',
+        'line-height': 'line-height',
```
Comment (smyzin) [inline tailwind.config.js:148]: Do we need it?

File: src/components/Announcement.tsx [hunk 4]
```diff
@@ -51,18 +60,26 @@ export default function ({ redirectTo }: { redirectTo: string }) {
   const onPage = location.pathname === redirectTo
 
   return (
-    <div id="bottom-bar" className={announceWrapper(animate)}>
+    <div className={announceWrapper(animate)}>
```
Comment (T-Damer) [inline src/components/Announcement.tsx:63]: Why we needed an `id` here?

File: src/components/Announcement.tsx [hunk 5]
```diff
@@ -51,18 +60,26 @@ export default function ({ redirectTo }: { redirectTo: string }) {
...
-          {!onPage
-            ? 'Get started'
-            : account
-            ? 'Get started below.'
-            : 'Connect wallet to get started.'}
-        </AccentText>
-      </LinkText>
+      <div
+        className={classnames(
+          opacity(animate ? 'opacity-0' : undefined),
```
Comment (T-Damer) [inline src/components/Announcement.tsx:67]: opacity-100 is default, let's do `{'opacity-0': animate}`

File: src/components/Announcement.tsx (hunk 4 above)
Comment (upacyxou) [reply src/components/Announcement.tsx:63]: Don't have any idea, it's not used anywhere in the code

File: src/components/Announcement.tsx (hunk 5 above)
Comment (T-Damer) [reply src/components/Announcement.tsx:67]: And I think we need to extract those classnames into separate `const`

File: src/components/Announcement.tsx [hunk 6]
```diff
@@ -51,18 +60,26 @@ export default function ({ redirectTo }: { redirectTo: string }) {
   const onPage = location.pathname === redirectTo
 
   return (
-    <div id="bottom-bar" className={announceWrapper(animate)}>
```
Comment (smyzin) [inline src/components/Announcement.tsx:54]: We don't need it anymore?

File: src/components/Announcement.tsx [hunk 7]
```diff
@@ -51,18 +60,26 @@ export default function ({ redirectTo }: { redirectTo: string }) {
...
+        className={classnames(
+          opacity(animate ? 'opacity-0' : undefined),
+          transitionProperty('transition-opacity'),
+          commonTransition
+        )}
+      >
+        <LinkText url={redirectTo}>
+          <AccentText small bold color="text-formal-accent">
+            Now introducing zk proofs for your work email!{' '}
+            {!onPage
```
Comment (T-Damer) [inline src/components/Announcement.tsx:75]: Let's correct the logic here:
```TS
onPage ? account ? 'Get started below.' : 'Connect wallet to get started' : 'Get started'
```
I think it's easier to read, you can extract central block like:
```TS
const hasAccount = account ? account ? 'Get started below.' : 'Connect wallet to get started'
onPage ? hasAccount : 'Get started'
```

**Review by T-Damer — COMMENTED** [review]
let's clean up a little :)

File: src/components/Announcement.tsx (hunk 6 above)
Comment (upacyxou) [reply src/components/Announcement.tsx:54]: It's not used in the code at all

File: src/components/Announcement.tsx (hunk 2 above)
Comment (upacyxou) [reply src/components/Announcement.tsx:43]: Got it, thank you

File: src/components/Announcement.tsx (hunk 7 above)
Comment (upacyxou) [reply src/components/Announcement.tsx:75]: Thank you! Fixed

File: tailwind.config.js (hunk 3 above)
Comment (upacyxou) [reply tailwind.config.js:148]: Not anymore, fixed, thank you

**Review by backmeupplz — APPROVED** [review]

**Pull request merged**